lib/zhr_devs/submissions/commands/submit_solution.ex

```diff
@@ -0,0 +1,100 @@
+defmodule ZhrDevs.Submissions.Commands.SubmitSolution do
+  @moduledoc """
+  A command must contain a field to uniquely identify the aggregate instance (e.g. account_number).
+  Use @enforce_keys to force the identity field to be specified when creating the command struct.
+  """
+  @fields [:uuid, :hashed_identity, :technology, :task_uuid, :solution_path, :submission_identity]
```

Comment:
Here we reuse part of these fields

Reply:
And the reason for this to be partial copy pasta is actually that events decouple the logic. As long as something expands to a map with these fields no matter from where we got it, it'll be interpreted as Solution submitted event.

Am I right?

Reply:
No. It can be seems like copy pasta, but unfortunately it isn't. The command fields come from scary internet, that's why we use a plain struct here. Then we indeed **reuse** the part of these fields, because without those fields it is impossible to achieve what we are trying to do.

Reply:
It's ok, I'm just a bit damaged by ecto. As long as we aren't writing bugs, we're good 😃